### DISCUSSION TITLE
CSV device import: Reference site by custom_field

### DISCUSSION THREAD

**Discussion by moetezz** [netbox-community/netbox#16114] [Q&A]
Hello
I am trying to import some devices using the csv import and I want to try to reference site by a custom field I have building record number as it is more common in our org to use rather than the site name but seems like it doesn't work.
Does anyone know? or if it is even possible here is an the csv header:
name,role,manufacturer,device_type,status,site.cf_building_record_number,cf_ap_group

**Comment by candlerb** [comment]
As far as I know the only options supported are `site.id` and `site` (name implied). Having said that, I *thought* there were a couple of test cases for it in the source code, and I can't find them.

You could try using the slug for your building record number. CSV import may or may not support `site.slug` as a column (try it), but even if it doesn't, it helps with filtering in the REST API: `/api/dcim/devices/?site=some-site-slug`

**Comment by moetezz** [comment]
as I tested with version 3.7.6, it is actually possible to reference site in the device import using `site.facility` for example but `site.cf_building_record_number` doesn't work. maybe I am doing something wrong, I haven't yet figure out, if anyone know how please share with us
